tests/trainer/test_trainer.py

```diff
@@ -1604,6 +1604,46 @@ def test_pytorch_profiler_nested_emit_nvtx(tmpdir):
     trainer.fit(model)
 
 
+def test_profiler_teardown(tmpdir):
```

Comment:
you could add a dummy profiler which sets profiler state in `teardown` which you can use to verify in the unit test that `teardown` is called

do we have a separate `test_profilers.py` ? i think it'd be useful to split out from the trainer. what do you think @tchaton ?

Reply:
> do we have a separate test_profilers.py ? i think it'd be useful to split out from the trainer

Yes! To `tests/test_profiler.py`

I have actually moved them in my (very) WIP PR to refactor profilers

Reply:
Moved in https://github.com/PyTorchLightning/pytorch-lightning/commit/51c9260fad5b1ed3b4e41a9ebf460bf4c609fe2a